src/main/java/task/Task.java

```diff
@@ -0,0 +1,73 @@
+package task;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a task that a user can add to, delete from, find, or set to done in a task list.
+ */
+public abstract class Task {
+    private final String taskName;
+    private boolean done;
```

Comment:
Perhaps can use isDone as boolean name since it is more intuitive as description of the current state? (Though done sounds ok also).